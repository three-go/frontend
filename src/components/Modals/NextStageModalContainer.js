import React, { useState } from "react";

import PropTypes from "prop-types";

import { ButtonModal, SmallButton } from "..";
import { colors } from "../../common";

const NextStageModalContainer = ({ onNextStage }) => {
  const [modalVisible, setModalVisible] = useState(true);

  const handleNextStage = () => {
    setModalVisible(false);
    onNextStage();
  };

  return (
    <ButtonModal isVisible={modalVisible} content="탈출에 성공하셨습니다.">
      <SmallButton
        content="다음 단계로"
        color={colors.tealGreen}
        onPress={handleNextStage}
      />
    </ButtonModal>
  );
};

export default NextStageModalContainer;

NextStageModalContainer.propTypes = {
  onNextStage: PropTypes.func,
};
