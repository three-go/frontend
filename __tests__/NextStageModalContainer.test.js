import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import NextStageModalContainer from "../src/components/Modals/NextStageModalContainer";

describe("NextStageModalContainer", () => {
  it("NextStageModalContainer render check", () => {
    const mockOnNextStage = jest.fn();

    const component = render(
      <NextStageModalContainer onNextStage={mockOnNextStage} />
    );

    expect(component.getByText("탈출에 성공하셨습니다.")).toBeTruthy();
    expect(component.getByText("다음 단계로")).toBeTruthy();
    expect(component.toJSON).toMatchSnapshot();
  });

  it("NextStageModalContainer '다음 단계로' button click", () => {
    const mockOnNextStage = jest.fn();

    const component = render(
      <NextStageModalContainer onNextStage={mockOnNextStage} />
    );

    const nextStageButton = component.getByText("다음 단계로");
    fireEvent.press(nextStageButton);

    expect(mockOnNextStage.mock.calls.length).toBe(1);
    expect(() => component.getByText("다음 단계로")).toThrow(
      "Unable to find an element with text: 다음 단계로"
    );
  });
});
