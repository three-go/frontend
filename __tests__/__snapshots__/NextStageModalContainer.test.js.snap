// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextStageModalContainer NextStageModalContainer render check 1`] = `[Function]`;
